dvf_json/src/Plugin/Visualisation/Source/JsonFile.php

```diff
@@ -0,0 +1,242 @@
+  protected $fields;
+
+  /**
+   * The JSON file records.
+   *
+   * @var array
+   */
+  protected $records;
+
+  /**
+   * Constructs a new JsonFile.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\dvf\Plugin\VisualisationInterface $visualisation
+   *   The visualisation context in which the plugin will run.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
+   *   The cache backend.
+   * @param \GuzzleHttp\Client $http_client
+   *   The HTTP client.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, VisualisationInterface $visualisation = NULL, ModuleHandlerInterface $module_handler, CacheBackendInterface $cache, Client $http_client) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $visualisation, $module_handler);
+    $this->cache = $cache;
+    $this->httpClient = $http_client;
+  }
+
+  /**
+   * Creates an instance of the plugin.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The container to pull out services used in the plugin.
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\dvf\Plugin\VisualisationInterface $visualisation
+   *   The visualisation context in which the plugin will run.
+   *
+   * @return static
+   *   Returns an instance of this plugin.
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, VisualisationInterface $visualisation = NULL) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $visualisation,
+      $container->get('module_handler'),
+      $container->get('cache.dvf_json'),
+      $container->get('http_client')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'json' => [
+        'expression' => '',
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $form['json'] = [
+      '#type' => 'details',
+      '#title' => $this->t('JSON file settings'),
+      '#open' => TRUE,
+      '#tree' => TRUE,
+    ];
+
+    $form['json']['expression'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Expression'),
+      '#description' => 'JSONPath expression used to extract the data (e.g. $.store.book[*]).',
```

Comment:
Is it worth linking to a resource for JSONPath? e.g. http://jsonpath.com/ https://github.com/FlowCommunications/JSONPath or http://goessner.net/articles/JsonPath/

Reply:
Absolutely! Just need to find the best way to document it as websites like jsonpath.com support a non-standard JSONPath syntax which includes extra features.

Reply:
This has been [fixed](https://github.com/govCMS/dvf/blob/feature/dvf-json/dvf_json/src/Plugin/Visualisation/Source/JsonFile.php#L134-L139). I have also added a nice [README](https://github.com/govCMS/dvf/tree/feature/dvf-json/dvf_json) file.